services/web/server/tests/unit/with_dbs/09/test_projects_01.py

```diff
@@ -450,108 +448,6 @@ async def _replace_project(
     return data
 
 
-async def _connect_websocket(
-    socketio_client: Callable,
-    check_connection: bool,
-    client,
-    client_id: str,
-    events: Optional[Dict[str, Callable]] = None,
-) -> socketio.AsyncClient:
-    try:
-        sio = await socketio_client(client_id, client)
-        assert sio.sid
-        if events:
-            for event, handler in events.items():
-                sio.on(event, handler=handler)
-        return sio
-    except SocketConnectionError:
-        if check_connection:
-            pytest.fail("socket io connection should not fail")
-
-
-async def _open_project(
-    client,
-    client_id: str,
-    project: Dict,
-    expected: Union[web.HTTPException, List[web.HTTPException]],
-) -> Optional[Tuple[Dict, Dict]]:
-    url = client.app.router["open_project"].url_for(project_id=project["uuid"])
-    resp = await client.post(url, json=client_id)
-
-    if isinstance(expected, list):
-        for e in expected:
-            try:
-                data, error = await assert_status(resp, e)
-                return data, error
-            except AssertionError:
-                # re-raies if last item
-                if e == expected[-1]:
-                    raise
-                continue
-    else:
-        return await assert_status(resp, expected)
-
-
-async def _close_project(
-    client, client_id: str, project: Dict, expected: web.HTTPException
-):
-    url = client.app.router["close_project"].url_for(project_id=project["uuid"])
-    resp = await client.post(url, json=client_id)
-    await assert_status(resp, expected)
-
-
-async def _state_project(
-    client,
-    project: Dict,
-    expected: web.HTTPException,
-    expected_project_state: ProjectState,
-):
-    url = client.app.router["state_project"].url_for(project_id=project["uuid"])
-    resp = await client.get(url)
-    data, error = await assert_status(resp, expected)
-    if not error:
-        # the project is locked
-        received_state = ProjectState(**data)
-        assert received_state == expected_project_state
-
-
-async def _assert_project_state_updated(
```

Comment:
do you have a tool to find the ones that are unused? or was that a pylint warning I did not see?

Reply:
pylance and my eye :-)